src/backend/distributed/planner/multi_planner.c

```diff
@@ -436,9 +473,10 @@ CreateDistributedPlan(PlannedStmt *localPlan, Query *originalQuery, Query *query
 
 	/*
 	 * As explained above, force planning costs to be unrealistically high if
-	 * query planning failed (possibly) due to prepared statement parameters.
+	 * query planning failed (possibly) due to prepared statement parameters or
+	 * if it is planned as a multi shard modify query.
 	 */
-	if (distributedPlan->planningError && hasUnresolvedParams)
+	if ((distributedPlan->planningError || multiShardModifyQuery) && hasUnresolvedParams)
```

Comment:
Passing a simple boolean flag as a pointer all the way down the planner tree is a little ugly. Is there a function we can write that can tell whether a query is a `MultiShardModifyPlan(distributedPlan)`?